lib/client.js

```diff
@@ -1616,6 +1616,79 @@ class BaseClient {
     return responseBody;
   }
 
+  async backchannelAuthentication(params = {}, { clientAssertionPayload } = {}) {
+    assertIssuerConfiguration(this.issuer, 'backchannel_authentication_endpoint');
+
+    const body = {
+      ...params,
+      client_id: this.client_id,
+    };
+
+    const response = await authenticatedPost.call(
+      this,
+      'backchannel_authentication',
+      {
+        responseType: 'json',
+        form: body,
+      },
+      { clientAssertionPayload, endpointAuthMethod: 'token' },
+    );
+
+    const responseBody = processResponse(response, { statusCode: 200 });
+
+    if (!('expires_in' in responseBody)) {
+      throw new RPError({
+        message: 'expected expires_in in Backchannel Authentication Successful Response',
+        response,
+      });
+    }
+    if (typeof responseBody.expires_in !== 'number') {
+      throw new RPError({
+        message: 'invalid expires_in value in Backchannel Authentication Successful Response',
+        response,
+      });
+    }
+    if (!('auth_req_id' in responseBody)) {
+      throw new RPError({
+        message: 'expected auth_req_id in Backchannel Authentication Successful Response',
+        response,
+      });
+    }
+    if (typeof responseBody.auth_req_id !== 'string') {
+      throw new RPError({
+        message: 'invalid auth_req_id value in Backchnanel Authentication Successful Response',
+        response,
+      });
+    }
+
+    return responseBody;
+  }
+
+  async pollBackchannelAthentication(
+    body,
+    checks = {},
+    { exchangeBody, clientAssertionPayload, DPoP } = {},
+  ) {
+    if (this.default_max_age && !checks.max_age) {
+      checks.max_age = this.default_max_age;
+    }
+
+    const tokenset = await this.grant(
+      {
+        ...exchangeBody,
+        grant_type: 'urn:openid:params:grant-type:ciba',
```

Comment:
shouldn't it go below to prevent modifying it?

Reply:
You can modify it if you want. The standard CIBA servers should require the grant_type to be the default one but someone may decide to implement a modification of it and we should support it if that's the case